src/client/connect/dns.rs

```diff
@@ -22,7 +22,7 @@ use crate::common::{Future, Never, Pin, Poll, Unpin, task};
 /// Resolve a hostname to a set of IP addresses.
 pub trait Resolve: Unpin {
     /// The set of IP addresses to try to connect to.
-    type Addrs: Iterator<Item=IpAddr>;
+    type Addrs: Iterator<Item=SocketAddr>;
```

Comment:
I think this is modifying a public trait, so it'd be really nice if this could be considered for 0.13 inclusion, otherwise we might need to wait for the next minor version to not break users.